centreon/src/Core/HostCategory/Application/UseCase/CreateHostCategory/CreateHostCategory.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+* Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* https://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* For more information : contact@centreon.com
+*
+*/
+
+declare(strict_types=1);
+
+namespace Core\HostCategory\Application\UseCase\CreateHostCategory;
+
+use Centreon\Domain\Contact\Contact;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Application\Common\UseCase\InvalidArgumentResponse;
+use Core\Application\Common\UseCase\PresenterInterface;
+use Core\HostCategory\Application\Exception\HostCategoryException;
+use Core\HostCategory\Application\Repository\ReadHostCategoryRepositoryInterface;
+use Core\HostCategory\Application\Repository\WriteHostCategoryRepositoryInterface;
+use Core\HostCategory\Application\UseCase\CreateHostCategory\CreateHostCategoryRequest;
+use Core\HostCategory\Domain\Model\HostCategory;
+use Core\HostCategory\Domain\Model\NewHostCategory;
+
+final class CreateHostCategory
```

Comment:
⚠️ Please check the "convention" about **`Add`** vs ~~`Create`~~ we used on other APIs.
👉🏻 **`AddHostCategory`**